modules/saku-policy-store/src/saku/dal_datalevin.clj

```diff
@@ -1,135 +1,140 @@
 (ns saku.dal-datalevin
-  (:require [datalevin.core :as d]
-            [saku.dal-interface :as interface]
-            [saku.schemas :as schemas]))
-
+  (:require
+    [clojure.set :as sets]
+    [datalevin.core :as d]
+    [saku.dal-interface :as dal]
+    [saku.dal-interface :as interface]))
 
 (def ^:private pattern '[* {:policy/statements [* {:statement/effect [*]}]}])
 
-
-(defn ^:private impl-get-policies [db drns]
-  (d/q '[:find [(pull ?p pattern) ...]
-         :in $ pattern [?drns ...]
-         :where
-         [?p :policy/drn ?drns]]
-       db
+(def policy-type->statement-key
+  {"Resource" :statement/identities
+   "Identity" :statement/resources})
+(def statement-key->policy-type (sets/map-invert policy-type->statement-key))
+
+(defn ^:private policies-by-type [db drns statement-attr]
```

Comment:
Most often we want to fetch policies by their type, so we compose the query as such.